src/nano_contracts/deserializer.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+
+import { bufferToHex, unpackToFloat, unpackToInt } from '../utils/buffer';
+
+
+class Deserializer {
```

Comment:
```suggestion
class NanoContractArgumentDeserializer {
```

Just to be more specific of the reason for this class

Reply:
Another issue I have with this class is that it could be just an util with the content of `deserializeFromType`.
Is there a reason we need it to be a class?

Reply:
I changed the export name in the lib.ts file to be consistent with what we did in the previous PR that had the `Serializer` class. What do you think?

Regarding having a class, at the beginning I wasn't using only the `deserializeFromType`, then the class was required. Now it's just something more organized, I don't see any downside of having it, do you?